### PR TITLE
Use safer options for randomized_svd

### PR DESCRIPTION
**Pull request by mpuerrer**
Prevent segfaults in multi-threaded LU decomposition, by switching to the QR decomposition as the normalizer for the power iteration algorithm used in randomized_svd().

The segfault happens in when applying the LU normalizer to `A @ Q` at https://github.com/scikit-learn/scikit-learn/blob/98ed9dc73/sklearn/utils/extmath.py#L335, not in the call to the standard SVD.

### CODE REVIEW

**Review by stephengreen — CHANGES_REQUESTED** [review]
Thanks! So with this change, can we remove the pinned scipy version? If so, please remove it from `pyproject.toml`.

**Comment by mpuerrer** [comment]
I have removed pinning scipy now.

**Review by stephengreen — APPROVED** [review]
Great, thanks for tracking this down!
